WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m

```diff
@@ -84,10 +85,15 @@ - (XCElementSnapshot *)fb_lastSnapshot
   if (uniqueTypes && [uniqueTypes count] == 1) {
     type = [uniqueTypes.firstObject intValue];
   }
-  [matchedElements addObjectsFromArray:[[self descendantsMatchingType:type] matchingPredicate:[FBPredicate predicateWithFormat:@"%K IN %@", FBStringify(XCUIElement, wdUID), matchedUids]].allElementsBoundByIndex];
+  XCUIElementQuery *query = [[self descendantsMatchingType:type] matchingPredicate:[FBPredicate predicateWithFormat:@"%K IN %@", FBStringify(XCUIElement, wdUID), matchedUids]];
+  if (1 == snapshots.count) {
```

Comment:
**snapshots** is the simple array, that we receive as method argument. No resolution is happening here.